content/post/codesent.md

```diff
@@ -0,0 +1,396 @@
+our model overfits after 5 epochs - that is, runs across the whole training set. This is a typical
+situation in recurrent neural networks. I should have used overfit reduction techniques,
+e.g. dropout, but I did not have time. So we pick the graph state on epoch 5 and export it.
+
+Let's see how BiDiSentiment performs on some answers to that StackOverflow questions.
+
+```
+go get -v gopkg.in/vmarkovtsev/BiDiSentiment.v1/...
+
+echo "When I wrote this, only God and I understood \
+what I was doing. Now, God only knows" | $GOPATH/bin/sentiment
+```
+
+Output (negative probability): 0.8803515 - the sentiment is clearly negative.
+
+```
+echo "sometimes I believe compiler ignores \
+all my comments" | $GOPATH/bin/sentiment
+```
+
+Output: 0.88705057. Again negative.
+
+```
+echo "drunk, fix later" | $GOPATH/bin/sentiment
+```
+
+Output: 0.08034721. Positive.
+
+### Comment sentiment analysis
+
+Let's sum everything up and describe how we run source code comment sentiment analysis:
+
+1. Install and run [bblfshd](https://doc.bblf.sh/user/getting-started.html#running-with-docker-recommended).
+2. Install [hercules](https://github.com/src-d/hercules/releases).
+3. Install [libtensorflow](https://www.tensorflow.org/install/install_go).
+
+Then we analyze any Git repository like this:
+
+```
+hercules --sentiment https://github.com/golang/go
+```
+
+We emulate the neutral sentiment class as the middle between positiveness and
+negativeness and exclude them from further analysis. The results are shown on the table.
+
+| Project               | Positive | Negative | Effective |
+|-----------------------|----------|----------|-----------|
+| [pygame/pygame](https://github.com/pygame/pygame)         | 93.4     | 98.2     | -4.8      |
+| [pallets/flask](https://github.com/pallets/flask)         | 23.6     | 31.9     | -8.3      |
+| [django/django](https://github.com/django/django)         | 462.1    | 536.1    | -74.0     |
+| [golang/go](https://github.com/golang/go)             | 402.1    | 507.3    | -105.2    |
+| [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) | 186.1    | 91.6     | 94.5      |
+| [keras-team/keras](https://github.com/keras-team/keras)      | 104.8    | 71.1     | 33.7      |
+
+The columns represent the sum of all the positive and negative sentiments together with the final
+sentiment value. Some of the projects appeared to be negative and some positive. We'll see why
+later, and now let's look at the promised sentiment plot through time.
+
+{{% caption src="/post/codesent/golang-sentiment-black.png" %}}
+golang/go sentiment through time.
+{{% /caption %}}
+
+Can we conclude anything useful from this? Unfortunately, no, there are no clear trends or
+periods. It is too many different comments. Maybe we will have more luck with monorepos, that is
+those with a single main contributor. For now, it is more fun to inspect the most negative and positive
```

Comment:
that's not what a monorepo is, a monorepo is a repository that has all of the code for a company, such as Google

avoid the term here, instead call it "single contributor repo" or something like that

Reply:
I used this a few times as "single contrib repo" before. As you've already discovered I like giving short names to frequent terms, and "monorepo" was one of them back in 2016-2017. No objections to changing it though.